src/app/lightning/update-ln-payments.ts

```diff
@@ -12,63 +12,80 @@ export const updateLnPayments = async (): Promise<true | ApplicationError> => {
 
   const lndService = LndService()
   if (lndService instanceof Error) return lndService
+
+  const listFns = [lndService.listSettledPayments, lndService.listFailedPayments]
   const pubkeys = lndService
     .listActivePubkeys()
     .filter((pubkey) => pubkeysFromPayments.has(pubkey))
 
-  for (const key of pubkeys) {
-    const pubkey = key as Pubkey
+  for (const listFn of listFns) {
+    for (const key of pubkeys) {
+      processedLnPaymentsHashes = await updateLnPaymentsByFunction({
+        processedLnPaymentsHashes,
+        incompleteLnPayments,
+        pubkey: key as Pubkey,
+        listFn,
+      })
+    }
+  }
+  return true
+}
 
-    let after: PagingStartToken | PagingContinueToken | PagingStopToken = undefined
-    while (
-      processedLnPaymentsHashes.length < incompleteLnPayments.length &&
-      after !== false
-    ) {
-      // Fetch from Lightning service
-      const results: ListLnPaymentsResult | LightningServiceError =
-        await lndService.listSettledAndFailedPayments({
-          pubkey,
-          after,
-        })
-      if (results instanceof Error) {
-        baseLogger.error(
-          { error: results },
-          `Could not fetch payments for pubkey ${pubkey}`,
-        )
-        break
-      }
-      if (after === results.endCursor) break
-      after = results.endCursor
+const updateLnPaymentsByFunction = async ({
+  processedLnPaymentsHashes,
+  incompleteLnPayments,
+  pubkey,
+  listFn,
+}): Promise<PaymentHash[]> => {
+  let after: PagingStartToken | PagingContinueToken | PagingStopToken = undefined
```

Comment:
I think this is the type of value that we could also show as an event with opentemeletry. 

like this we would get a clear sense of the evolution of this longer running loop from honeycomb 

could be something like `addEventToCurrentSpan("updateLnPayments", {after})`

Reply:
Done: [sample trace](https://ui.honeycomb.io/galoy/datasets/arvin-dev/result/rg6zNdKvS2D/trace/iUcHXZVbfhZ?span=1ee6b89ae9d5bdfb)